.github/workflows/nightly_nosmp.yml

```diff
@@ -0,0 +1,119 @@
+name: Tests with SMP support (nightly)
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - staging
+      - trying
+
+jobs:
+  build:
+    name: Tests without SMP support
+    runs-on: ${{ matrix.os }}
+    # these tests based on the nightly compiler, which can be broken
+    # consequently, we continue on an error
+    #continue-on-error: true
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        rust: [nightly]
+        include:
+          - os: macOS-latest
+            rust: 'nightly'
+            components: 'rust-src, llvm-tools-preview'
+            targets: 'x86_64-apple-darwin'
+          - os: windows-latest
+            rust: 'nightly'
+            components: 'rust-src, llvm-tools-preview'
+            targets: 'x86_64-pc-windows-msvc'
+          - os: ubuntu-latest
+            rust: 'nightly'
+            components: 'rust-src, llvm-tools-preview'
+            targets: 'x86_64-unknown-linux-gnu'
+
+
+    steps:
+      - uses: hecrj/setup-rust-action@v1
+        with:
+          rust-version: ${{ matrix.rust }}
+          components: ${{ matrix.components || '' }}
+          targets: ${{ matrix.targets || '' }}
+      - name: Checkout rusty-hermit
+        uses: actions/checkout@v2
+        with:
+          repository: 'hermitcore/rusty-hermit'
+          ref: 'master'
+          submodules: 'true'
+      - name: Remove submodule libhermit-rs
+        run: rm -r libhermit-rs
+        if: ${{ ( matrix.os == 'macOS-latest' ) || ( matrix.os == 'ubuntu-latest' ) }}
+      - name: Remove submodule libhermit-rs (windows)
+        run: git rm -r libhermit-rs
+        if: ${{ matrix.os == 'windows-latest' }}
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+        with:
+          path: 'libhermit-rs'
+      - name: Check Cargo availability
+        run: cargo --version
+      - name: Cargo Test libhermit-rs (Unittests on Host)
+        run: cargo test --lib --target x86_64-unknown-linux-gnu
+        working-directory: libhermit-rs
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+      - name: Install qemu/nasm (apt)
+        run: sudo apt-get update --fix-missing && sudo apt-get install qemu-system-x86 nasm
```

Comment:
Is nasm still required?

Reply:
`nasm` is required to build the boot loader, which we need to run the kernel within qemu. The unikernel doesn't depend on `nasm`.